### PR TITLE
Sync Algorand's transactions

### PR DESCRIPTION
**Pull request by Riddlerrr**
Task: https://www.pivotaltracker.com/story/show/175845429

In this PR I've added sync for ALGO transactions only.
Assets transactions will be implemented separately.

### CODE REVIEW

File: app/models/blockchain/algorand.rb [hunk 1]
```diff
@@ -24,6 +24,10 @@ def asset_api_path(asset_id)
     "/v2/assets/#{asset_id}"
   end
 
+  def get_transaction_path(tx_hash)
```
Comment (Alexey1100) [inline app/models/blockchain/algorand.rb:27]: Looks like we already have it:

Comment (Riddlerrr) [reply app/models/blockchain/algorand.rb:27]: 👍

File: app/lib/comakery/algorand/tx.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    transaction_data.dig('payment-transaction', 'receiver') || ''
+  end
+
+  # In MicroAlgos for ALGO transaction
+  def amount
+    transaction_data.dig('payment-transaction', 'amount') || 0
+  end
+
+  def confirmed?(_number_of_confirmations = 1)
+    confirmed_round.present? && current_round > confirmed_round
```
Comment (aquabu) [inline app/lib/comakery/algorand/tx.rb:43]: perhaps `current_round >= confirmed_round`
number of confirmations is ignored due to blockchain finality on Algorand?

File: app/lib/comakery/algorand/tx.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  end
+
+  def valid?(blockchain_transaction)
+    return false unless confirmed?
+    return false unless valid_addresses?(blockchain_transaction)
+    return false unless valid_amount?(blockchain_transaction)
+
```
Comment (aquabu) [inline app/lib/comakery/algorand/tx.rb:50]: May want an additional check here to verify the database stored blockchain_transaction current block is < confirmed round on the blockchain transaction. This is to verify that the transaction is is in a later block on the blockchain then when the payment interaction started on CoMakery.

**Review by aquabu — APPROVED** [review]

**Pull request merged**